sonarr/series.go

```diff
@@ -6,11 +6,89 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/url"
+	"path"
 	"strconv"
+	"time"
 
 	"golift.io/starr"
 )
 
+// Series the /api/v3/series endpoint.
+type Series struct {
+	Ended             bool              `json:"ended,omitempty"`
+	Monitored         bool              `json:"monitored"`
+	SeasonFolder      bool              `json:"seasonFolder,omitempty"`
+	UseSceneNumbering bool              `json:"useSceneNumbering,omitempty"`
+	Runtime           int               `json:"runtime,omitempty"`
+	Year              int               `json:"year,omitempty"`
+	ID                int64             `json:"id,omitempty"`
+	LanguageProfileID int64             `json:"languageProfileId,omitempty"`
+	QualityProfileID  int64             `json:"qualityProfileId,omitempty"`
+	TvdbID            int64             `json:"tvdbId,omitempty"`
+	TvMazeID          int64             `json:"tvMazeId,omitempty"`
+	TvRageID          int64             `json:"tvRageId,omitempty"`
+	AirTime           string            `json:"airTime,omitempty"`
+	Certification     string            `json:"certification,omitempty"`
+	CleanTitle        string            `json:"cleanTitle,omitempty"`
+	ImdbID            string            `json:"imdbId,omitempty"`
+	Network           string            `json:"network,omitempty"`
+	Overview          string            `json:"overview,omitempty"`
+	Path              string            `json:"path,omitempty"`
+	SeriesType        string            `json:"seriesType,omitempty"`
+	SortTitle         string            `json:"sortTitle,omitempty"`
+	Status            string            `json:"status,omitempty"`
+	Title             string            `json:"title,omitempty"`
+	TitleSlug         string            `json:"titleSlug,omitempty"`
+	RootFolderPath    string            `json:"rootFolderPath,omitempty"`
+	Added             *time.Time        `json:"added,omitempty"`
+	FirstAired        *time.Time        `json:"firstAired,omitempty"`
+	NextAiring        *time.Time        `json:"nextAiring,omitempty"`
+	PreviousAiring    *time.Time        `json:"previousAiring,omitempty"`
```

Comment:
These are all outputs, and making them pointers has no value.

Reply:
If the struct is the same for input and output, than having a pointer will allow to omit them in the request

Reply:
Is it necessary though? The apps may ignore these if you try to add them. 

But also, this goes with the problem above. This is _a lot of change_ to make these methods "reuse" the same data. When it makes sense to make things line up, I do it, but I've avoided overloading inputs with large outputs because it requires concessions like this, and for what actual gain?*  When you use things like AWS Go library, these "different" inputs and outputs seem to make sense.

* You mentioned being able to copy an output directly to an input to add something. I assume this means "copying from one system to another." This is a sane argument, but you're expecting end users to have some idea which fields and needed and which aren't when they go to try to change things. When they see a date like `Added` set to "something" and wonder "do I need to change that" it becomes awkward.

Reply:
I was referring to the chance to take the output of a `Lookup` method and pass it directly to the `AddSeries` method.
Something like:
```go
lookupSeries, _ := client.GetSeriesLookup("", tvdbID)
series, _ := client.AddSeries(lookupSeries[0])
```
I haven't tried it yet, but it should work.
This way, a user could be able to add a series just by knowing its tvdbID, or even just the name (if they want to make some other checks).

I'd say that this is the main advantage of having a single struct. A user can onboard even without knowing the meaning of the fields. Maybe I have been too aggressive with the `omitempty` tag, it could be used to identify which parameters are actually mandatory.

About the AWS library, I think that they are following a different approach. As far as I understand, they are defining for each api call both input and output, reusing only the nested types. 
Then, for each struct they define different methods, like validators, when needed. If you want to follow this approach, it's fine but then we should probably do that for each api call.

Reply:
If the goal is this 
```go
lookupSeries, _ := client.GetSeriesLookup("", tvdbID) 
series, _ := client.AddSeries(lookupSeries[0])
```

You could just add another method. `AddSeriesByTvDbID(tvdbID)`

Reply:
Here I am, sorry for the late reply, busy couple of weeks.
The goal was mainly to be consistent across all resources. the easier management between methods would've been just a plus.
However, If you like to have different structs between input and output, it's fine for me. I'll try to change this PR accordingly. And I'll keep that in mind for other resources as well (even the profiles I added before should probably be changed accordingly).
Unfortunately, it will take some time, since I'll have to manually test each field if it can be a valid input or not.
I'll be back with the changes!
thanks

Reply:
Hi @davidnewhall, I made few changes at this PR to have different structs between input and output as you requested.
- I changed back the time.Time to value instead of pointer
- The input is the same for both POST and PUT, do you think we should separate the 2 of them due to AddSeriesOption?
Let me know if you prefer a different naming

Reply:
I recently started using the `main` branch in notifiarr, and having to switch addTag and updateTag to use structs as input really sucked and didn't make any sense. Just wanted to put that out there. Thanks for updating this PR, I'll take a look shortly.

Reply:
> However, If you like to have different structs between input and output, it's fine for me. 

This is just how the Starr apps are. I want to be consistent with _them_.

> The input is the same for both POST and PUT, do you think we should separate the 2 of them

No.

> Let me know if you prefer a different naming

I left comments about the struct names. Once you fix those I'll test again, and make sure things are still working as expected.

Reply:
Ok, Just committed the changes and rebased, let me know if you have other feedbacks

Reply:
Would be easier to review the changes if you hadn't force-pushed. :)

Reply:
sorry, my bad! I'll try to keep in mind that I have to rebase only at the end!